### PR TITLE
Load module types from referenced group for app_policies

### LINKED ISSUE #2670: SDL process RC requests with resultCode DISALLOWED after IGN_OFF and IGN_ON
[Delete any non-applicable sections, but we may ask for more information.]

### Bug Report
SDL process GetInteriorVehicleData with resultCode DISALLOWED after IGN_OFF and IGN_ON

##### Reproduction Steps
1. Define module type module_1 and in default section in preloaded_pt.json
2. Update groups in default with "RemoteControl" group
3. Register RC app
4. App requests successful GetInteriorVD with module_1
5. Perform IGN_OFF
6. Perform IGN_ON
7. Register same RC app
8. Request GetInteriorVD with allowed module_1

##### Expected Behavior
SDL process  GetInteriorVD with allowed module_1 successful and resend request to HMI

##### Observed Behavior
SDL responds with DISALLOWED resultCode to GetInteriorVD

##### OS & Version Information
* OS/Version: Ubuntu 16.04
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/commit/8da2a76d7dbde7845c1116034f8a60490f4fad0c
* HMI version: https://github.com/smartdevicelink/sdl_hmi/commit/51a4ace4be6bab9613de87d3831522ad443acd68

SDL logs: 
[logs.tar.gz](https://github.com/smartdevicelink/sdl_core/files/2469848/logs.tar.gz)
Script for reproducing: 
[GetInteriorVD_after_ignition_off.lua.tar.gz](https://github.com/smartdevicelink/sdl_core/files/2470266/GetInteriorVD_after_ignition_off.lua.tar.gz)

ATF: https://github.com/smartdevicelink/sdl_atf/commit/c7022ed1aa9cec3a3ae5ceff86cc7690a4df7c14
Scripts: https://github.com/smartdevicelink/sdl_atf_test_scripts/commit/48d2a16eb73133cae7079a8457f3becb624ca23f

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Fixes #2670

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Tested with script attached to #2670

### Summary
If application have default policies, sqldb does not store module types for this application separately.

During loading db SDL should check that application have default policies
and select from module_types sql table sing  "default" key but not appid key.

In function `GatherApplicationPoliciesSection` already exist check for default policies :

```
    if (IsDefaultPolicy(app_id)) {
      (*policies).apps[app_id].set_to_string(kDefaultId);
    }
```

Its setup stringified representation of application policies : "default".

This stringified reference ("default") used for select module type.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@LuxoftAKutsan check code style please

File: src/components/policy/policy_regular/src/sql_pt_representation.cc [hunk 1]
```diff
@@ -728,7 +728,9 @@ bool SQLPTRepresentation::GatherApplicationPoliciesSection(
       return false;
     }
     if (!denied) {
-      if (!GatherModuleType(app_id, &*params.moduleType)) {
+      const auto& module_type_app_id = (*policies).apps[app_id].get_string();
```
[the remark this answers is not in the document]
Comment (jacobkeeler) [reply src/components/policy/policy_regular/src/sql_pt_representation.cc]: Also, most of SetDefaultPolicy would be redundant with this change, I believe everything after `SetPreloaded(false);` can be deleted.

[the remark this answers is not in the document]
Comment (LuxoftAKutsan) [reply src/components/policy/policy_regular/src/sql_pt_representation.cc]: @jacobkeeler  what do you mean that everything after SetPreloaded(false); ?

[the remark this answers is not in the document]
Comment (jacobkeeler) [reply src/components/policy/policy_regular/src/sql_pt_representation.cc]: @LuxoftAKutsan In reference to the contents of `SetDefaultPolicy`: https://github.com/smartdevicelink/sdl_core/blob/c21161857f282f5b3bf31cb9db18e5d312a898fa/src/components/policy/policy_regular/src/sql_pt_representation.cc#L1941

[the remark this answers is not in the document]
Comment (LuxoftAKutsan) [reply src/components/policy/policy_regular/src/sql_pt_representation.cc]: @jacobkeeler fixed according to comment. 

SetDefaultPolicies also contains code related to Requests subtypes ( inserting additional fields to data base for further db loading). I think that this is redundant duplication of data, but not sure that I should fix in in scope of this PR. Should I?

**Comment by jacobkeeler** [comment]
Sorry if it was unclear, but the exact change I was looking for was:
```
     const auto& gather_app_id = ((*policies).apps[app_id].is_string())
                                     ? (*policies).apps[app_id].get_string()
                                     : app_id;
     if (!GatherAppGroup(gather_app_id, &params.groups)) {
       return false;
     }
     bool denied = false;
     if (!GatherRemoteControlDenied(gather_app_id, &denied)) {
       return false;
     }
     if (!denied) {
       // Module types should be selected from db by referenced groups
       if (!GatherModuleType(gather_app_id, &*params.moduleType)) {
         return false;
       }
     }
     if (!GatherNickName(gather_app_id, &*params.nicknames)) {
       return false;
     }
     if (!GatherAppType(gather_app_id, &*params.AppHMIType)) {
       return false;
     }
     if (!GatherRequestType(gather_app_id, &*params.RequestType)) {
       return false;
     }
 
     if (!GatherRequestSubType(gather_app_id, &*params.RequestSubType)) {
       return false;
     }
```
This means that the default policies will be loaded for this app if it has default, instead of that app's individual policies.

**Comment by LuxoftAKutsan** [comment]
@jacobkeeler  fixed according to commits

**Comment by jacobkeeler** [comment]
@LuxoftAKutsan looks like this bug also affects EXTERNAL_PROPRIETARY, so the fix will need to be applied to https://github.com/smartdevicelink/sdl_core/blob/fc8bd80fe92d1ebf5fcfe023ec000a03348b8208/src/components/policy/policy_external/src/sql_pt_ext_representation.cc as well

**Comment by LuxoftAKutsan** [comment]
@jacobkeeler added fix for external.

**Comment by jacobkeeler** [comment]
@LuxoftAKutsan fix style

**Comment by LuxoftAKutsan** [comment]
@jacobkeeler  fixed typos, reordered commits, removed additional unused files.

**Pull request merged**